tests/extra_source_and_tests/extra_src/shared/pressure_boundary/bidirectional_buffer.h

```diff
@@ -0,0 +1,155 @@
+/* ------------------------------------------------------------------------- *
+ *                                SPHinXsys                                  *
+ * ------------------------------------------------------------------------- *
+ * SPHinXsys (pronunciation: s'finksis) is an acronym from Smoothed Particle *
+ * Hydrodynamics for industrial compleX systems. It provides C++ APIs for    *
+ * physical accurate simulation and aims to model coupled industrial dynamic *
+ * systems including fluid, solid, multi-body dynamics and beyond with SPH   *
+ * (smoothed particle hydrodynamics), a meshless computational method using  *
+ * particle discretization.                                                  *
+ *                                                                           *
+ * SPHinXsys is partially funded by German Research Foundation               *
+ * (Deutsche Forschungsgemeinschaft) DFG HU1527/6-1, HU1527/10-1,            *
+ *  HU1527/12-1 and HU1527/12-4.                                             *
+ *                                                                           *
+ * Portions copyright (c) 2017-2023 Technical University of Munich and       *
+ * the authors' affiliations.                                                *
+ *                                                                           *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may   *
+ * not use this file except in compliance with the License. You may obtain a *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.        *
+ *                                                                           *
+ * ------------------------------------------------------------------------- */
+/**
+ * @file 	fluid_dynamics_inner.h
+ * @brief 	Here, we define the algorithm classes for fluid dynamics within the body.
+ * @details We consider here weakly compressible fluids.
+ * 			Note that, as these are local dynamics which are combined with particle dynamics
+ * 			algorithms as template, the name-hiding is used for functions in the derived classes.
+ * @author	Chi Zhang and Xiangyu Hu
+ */
+
+#ifndef BIDIRECTIONAL_BUFFER_H
+#define BIDIRECTIONAL_BUFFER_H
+
+#include "sphinxsys.h"
+
+namespace SPH
+{
+namespace fluid_dynamics
+{
+class BidirectionalBuffer
+{
+  protected:
+    ConcurrentIndexVector buffer_particle_list;
```

Comment:
member variable ended with under line "_', for example buffer_particle_list_;

Reply:
ok, i will revise it